webknossos-datastore/app/com/scalableminds/webknossos/datastore/services/SegmentIndexFileService.scala

```diff
@@ -0,0 +1,189 @@
+package com.scalableminds.webknossos.datastore.services
+
+import com.scalableminds.util.geometry.{BoundingBox, Vec3Int}
+import com.scalableminds.util.io.PathUtils
+import com.scalableminds.util.tools.{Fox, FoxImplicits}
+import com.scalableminds.webknossos.datastore.DataStoreConfig
+import com.scalableminds.webknossos.datastore.geometry.ListOfVec3IntProto
+import com.scalableminds.webknossos.datastore.helpers.SegmentStatistics
+import com.scalableminds.webknossos.datastore.models.datasource.DataLayer
+import com.scalableminds.webknossos.datastore.models.requests.{
+  Cuboid,
+  DataServiceDataRequest,
+  DataServiceRequestSettings
+}
+import com.scalableminds.webknossos.datastore.models.{
+  AdditionalCoordinate,
+  UnsignedInteger,
+  UnsignedIntegerArray,
+  VoxelPosition,
+  datasource
+}
+import com.scalableminds.webknossos.datastore.storage.{CachedHdf5File, Hdf5FileCache}
+import net.liftweb.common.{Box, Full}
+import play.api.i18n.MessagesProvider
+
+import java.nio.file.{Path, Paths}
+import javax.inject.Inject
+import scala.concurrent.ExecutionContext
+
+class SegmentIndexFileService @Inject()(config: DataStoreConfig,
+                                        binaryDataServiceHolder: BinaryDataServiceHolder,
+                                        dataSourceRepository: DataSourceRepository)(implicit ec: ExecutionContext)
+    extends FoxImplicits
+    with Hdf5HashedArrayUtils
+    with SegmentStatistics {
+  private val dataBaseDir = Paths.get(config.Datastore.baseFolder)
+  private val segmentIndexDir = "segmentIndex"
+
+  private lazy val fileHandleCache = new Hdf5FileCache(10)
+
+  def getSegmentIndexFile(organizationName: String, datasetName: String, dataLayerName: String): Box[Path] =
+    for {
+      _ <- Full("")
+      layerDir = dataBaseDir.resolve(organizationName).resolve(datasetName).resolve(dataLayerName)
+      segmentIndexDir = layerDir.resolve(this.segmentIndexDir)
+      files <- PathUtils.listFiles(segmentIndexDir, silent = true, PathUtils.fileExtensionFilter(hdf5FileExtension))
+      file <- files.headOption
+    } yield file
+
+  def readSegmentIndex(organizationName: String,
```

Comment:
here, too, I think a comment explaining which system the returned Vec3Ints are in would help future readers